### PR TITLE
mobile: add `EngineBuilder.useLegacyBuilder()` to Swift

### PR DESCRIPTION
**Pull request by jpsim**
Commit Message: mobile: add `EngineBuilder.useLegacyBuilder()` to Swift
Additional Description:
Risk Level: Medium to Envoy Mobile for ObjC/Swift.
Testing: Updated unit tests and added this config to the experimental app.
Docs Changes: None.
Release Notes: None.
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: mobile/library/objective-c/EnvoyEngineImpl.mm [hunk 1]
```diff
@@ -521,17 +522,43 @@ - (int)runWithTemplate:(NSString *)yaml
...
+              logLevel:(NSString *)logLevel {
+  NSString *resolvedYAML = [config resolveTemplate:yaml];
+  if (resolvedYAML == nil) {
+    return kEnvoyFailure;
+  }
+
+  [self performRegistrationsForConfig:config];
   [self startObservingLifecycleNotifications];
 
-  // Envoy exceptions will only be caught here when compiled for 64-bit arches.
```
Comment (jpsim) [inline mobile/library/objective-c/EnvoyEngineImpl.mm:531]: This comment is no longer relevant as we don't compile for any architectures other than 64-bit.

File: mobile/test/swift/integration/DirectResponseContainsHeadersMatchIntegrationTest.swift [hunk 2]
```diff
@@ -20,6 +20,7 @@ final class DirectResponseContainsHeadersMatchIntegrationTest: XCTestCase {
     .build()
 
     let engine = TestEngineBuilder()
+      .useLegacyBuilder(true)
```
Comment (alyssawilk) [inline mobile/test/swift/integration/DirectResponseContainsHeadersMatchIntegrationTest.swift:23]: just for my own awareness, are these tests not cleaned up yet, or are you just trying to ensure coverage for old and new?

Comment (jpsim) [reply mobile/test/swift/integration/DirectResponseContainsHeadersMatchIntegrationTest.swift:23]: The C++ builders don't support direct responses (yet).

I added support for that locally, and I should be able to put up a PR for discussion today or tomorrow.

**Review by alyssawilk — APPROVED** [review]

File: mobile/test/swift/integration/DirectResponseContainsHeadersMatchIntegrationTest.swift (hunk 2 above)
Comment (jpsim) [reply mobile/test/swift/integration/DirectResponseContainsHeadersMatchIntegrationTest.swift:23]: https://github.com/envoyproxy/envoy/pull/25552

**Pull request merged**